### PR TITLE
Added an action configuration validator for the coming actions + engine

### PR DESCRIPTION
**Pull request by mwwolters**
Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](http://forsetisecurity.org/docs/development/#executing-tests) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/forseti-security/854)

### CODE REVIEW

File: google/cloud/security/actions/action_config_validator.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+          Tuple: A tuple of the actions and errors.
+        """
+        action_ids = defaultdict(int)
+        actions = []
+        errors = []
+        for action in config.get('actions', []):
+            action_id = action.get('id')
+            if action_id is None:
+                errors.append(MissingRequiredActionField('id'))
+            action_ids[action_id] += 1
```
Comment (ahoying) [inline google/cloud/security/actions/action_config_validator.py]: I recommend making action_ids a set and then you change check if action_id in action_ids: add error.

File: google/cloud/security/actions/action_config_validator.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                return TriggerDoesntExist(trigger)
+            # TODO: once the rules classes and the action classes are set up,
+            # this should be changed handle any valid type instead of straight
+            # replacement.
+            parts[0] = 'google.cloud.security.auditor.rules'
+            try:
+                module = importlib.import_module('.'.join(parts[:-1]))
+                _ = getattr(module, parts[-1])
+            except (AttributeError, ImportError):
+                return TriggerDoesntExist(trigger)
```
Comment (ahoying) [inline google/cloud/security/actions/action_config_validator.py]: If there are multiple triggers that don't exist in a single action, the caller will only get the first one. This should return a list of all missing triggers that extends errors in validate. Be sure to update test_check_trigger_errors to check this.

File: google/cloud/security/actions/action_config_validator.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+        for action in actions:
+            action_errs = []
+            action_errs.append(
+                ActionsConfigValidator._check_action_type(action))
+            action_errs.append(ActionsConfigValidator._check_trigger(action))
+            for err in action_errs:
+                if err is not None:
+                    errors.append(err)
+        return config, errors
```
Comment (ahoying) [inline google/cloud/security/actions/action_config_validator.py]: Does it make sense to use a go idiom here? I would except an exception with the list of errors as an attribute.

**Comment by mwwolters** [comment]
Borked this, moved change to https://github.com/GoogleCloudPlatform/forseti-security/pull/857.

**Pull request closed without merging**